### PR TITLE
fetch ghana translation target languages from api

### PR DESCRIPTION
**Pull request by anish-work**
### Q/A checklist

- [x] If you add new dependencies, did you update the lock file?
```bash
poetry lock --no-update
```
- [x] Run tests 
```bash
ulimit -n unlimited && ./scripts/run-tests.sh
```
- [x] Do a self code review of the changes - Read the diff at least twice.  
- [x] Carefully think about the stuff that might break because of this change - this sounds obvious but it's easy to forget to do "Go to references" on each function you're changing and see if it's used in a way you didn't expect. 
- [x] The relevant pages still run when you press submit
- [x] The API for those pages still work (API tab)
- [x] The public API interface doesn't change if you didn't want it to (check API tab > docs page)
- [x] Do your UI changes (if applicable) look acceptable on mobile?
- [x] Ensure you have not regressed the import time unless you have a good reason to do so. 
You can visualize this using tuna:
```bash
python3 -X importtime -c 'import server' 2> out.log && tuna out.log
```
To measure import time for a specific library:
```bash
$ time python -c 'import pandas'

________________________________________________________
Executed in    1.15 secs    fish           external
   usr time    2.22 secs   86.00 micros    2.22 secs
   sys time    0.72 secs  613.00 micros    0.72 secs
```
To reduce import times, import libraries that take a long time inside the functions that use them instead of at the top of the file:
```python
def my_function():
    import pandas as pd
    ...
```

### Legal Boilerplate

Look, I get it. The entity doing business as “Gooey.AI” and/or “Dara.network” was incorporated in the State of Delaware in 2020 as Dara Network Inc. and is gonna need some rights from me in order to utilize my contributions in this PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Dara Network Inc can use, modify, copy, and redistribute my contributions, under its choice of terms.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208208561403730

### CODE REVIEW

File: daras_ai_v2/asr.py [hunk 1]
```diff
@@ -373,6 +373,21 @@ def google_translate_language_selector(
...
+@redis_cache_decorator(ex=settings.REDIS_MODELS_CACHE_EXPIRY)
+def ghana_nlp_translate_target_languages():
+    """
+    Get list of supported languages for Ghana NLP Translation.
+    :return: Dictionary of language codes and display names.
+    Reference: https://translation.ghananlp.org/api-details#api=ghananlp-translation-webservice-api
+    """
+    r = requests.get(
+        "https://translation-api.ghananlp.org/v1/languages",
+        headers={"Ocp-Apim-Subscription-Key": str(settings.GHANA_NLP_SUBKEY)},
```
Comment (devxpy) [inline daras_ai_v2/asr.py:385]: I think this is defined twice in the code, please refactor into 
`GHANA_API_AUTH_HEADERS = {"Ocp-Apim-Subscription-Key": str(settings.GHANA_NLP_SUBKEY)}`

File: daras_ai_v2/asr.py [hunk 2]
```diff
@@ -373,6 +373,21 @@ def google_translate_language_selector(
...
+    )
+    raise_for_status(r)
+    return r.json().get("languages", {})
```
Comment (devxpy) [inline daras_ai_v2/asr.py:388]: `r.json().get("languages")` or {} for null safety

Comment (anish-work) [reply daras_ai_v2/asr.py]: @devxpy changes made

File: daras_ai_v2/asr.py [hunk 3]
```diff
@@ -373,6 +374,21 @@ def google_translate_language_selector(
...
+    Get list of supported languages for Ghana NLP Translation.
+    :return: Dictionary of language codes and display names.
+    Reference: https://translation.ghananlp.org/api-details#api=ghananlp-translation-webservice-api
+    """
+    r = requests.get(
+        "https://translation-api.ghananlp.org/v1/languages",
+        headers=GHANA_API_AUTH_HEADERS,
+    )
+    raise_for_status(r)
+    return r.json().get("languages", {}) or {}
```
Comment (devxpy) [inline daras_ai_v2/asr.py:389]: just this is enough, the `, {}` part is redundant
`r.json().get("languages") or {}`

**Comment by anish-work** [comment]
@devxpy we have to update the env var for this to work

**Comment by devxpy** [comment]
lgtm

**Comment by devxpy** [comment]
please fix the tests above

**Comment by devxpy** [comment]
Eg here's what we did for azure https://github.com/GooeyAI/gooey-server/commit/85874ee1cbedf66803cc1d50a7fc6ee28db071d6

**Comment by anish-work** [comment]
@devxpy updated please check once

**Comment by devxpy** [comment]
better to at least test the render of the component by passing the available languages as empty like so https://github.com/GooeyAI/gooey-server/blob/d61653845dfabe99b56fc8cec17791f7518317cc/daras_ai_v2/text_to_speech_settings_widgets.py#L211-L223

**Comment by devxpy** [comment]
LGTM, but after testing: 
- doc extract (eg https://gooey.ai/doc-extract/direct-upload-kolb9ypq/ )
- audio input in /copilot as a input prompt
- audio input in /copilot as a RAG document
- audio input in /doc-search as a RAG document
- asr page itself

**Pull request merged**